packages/front-end/components/Metrics/MetricForm/index.tsx

```diff
@@ -362,368 +365,182 @@ const MetricForm: FC<MetricFormProps> = ({
   useEffect(() => {
     if (type === "binomial") {
       form.setValue("ignoreNulls", false);
+    } else {
+      requiredColumns.add("value");
```

Comment:
You can't change a set like this with `useEffect`.  Instead, move this logic into the `useMemo` call above.

```ts
  const requiredColumns = useMemo(() => {
    const set = new Set(["timestamp", ...value.userIdTypes]);
    if (type !== "binomial") {
      set.add("value");
    }
    return set;
  }, [value.userIdTypes, type]);
```